contracts/multisigs/SafeGuardFactory.sol

```diff
@@ -0,0 +1,144 @@
+    /// @notice A multisig has to have at least three owners to set up a safe guard.
+    /// @param payload Custom payload containing at least address and a function signature for a delegatecall.
+    function createSafeGuard(bytes memory payload) external {
+        // Check for the delegatecall
+        if (address(this) == factoryInstance) {
+            revert DelegateCallOnly(factoryInstance);
+        }
+
+        // Get the number of Safe owners
+        address[] memory owners = IGnosisSafe(address(this)).getOwners();
+        // Check for the number of owners
+        uint256 numOwners = owners.length;
+        if (numOwners < 3) {
+            revert InsufficientNumberOfOwners(numOwners, 3);
+        }
+
+        // Check for the payload length
+        uint256 payloadLength = payload.length;
+        if (payloadLength > 0 && payloadLength < DEFAULT_PAYLOAD_LENGTH) {
+            revert IncorrectPayloadLength(DEFAULT_PAYLOAD_LENGTH, payloadLength);
+        }
+
+        // Check that the guard slot is empty
+        bytes32 slot = GUARD_STORAGE_SLOT;
+        address guard;
+        // solhint-disable-next-line no-inline-assembly
+        assembly {
+            guard := sload(slot)
+        }
+        if (guard != address(0)) {
+            revert SafeGuardALreadyExists(guard);
+        }
+
+        // Execute the payload, if provided
+        // The payload must be executed before the safe guard is set, as the payload could potentially break the guard
+        if (payloadLength > 0) {
+            address target;
+            bool success;
+            // solhint-disable-next-line no-inline-assembly
+            assembly {
+                // First 20 bytes of the payload is address
+                target := mload(add(payload, 20))
+                // Offset 52 bytes from the payload start: initial 32 bytes plus the target address 20 bytes
+                success := delegatecall(gas(), target, add(payload, 52), mload(payload), 0, 0)
+            }
+            if (!success) {
+                revert PayloadExecFailed(payload);
+            }
+        }
+
+        // Bytecode of the Safe Guard contract
+        bytes memory bytecode = abi.encodePacked(type(SafeGuard).creationCode);
+        // Supposedly the address(this) is the unique address of a safe contract that calls this guard creation function
+        // Supposedly the same Safe contract is not going to create a guard, delete it and try to create again
+        // in the same block. Thus, the combination of the Safe address and the block timestamp will be unique
+        bytes32 salt = keccak256(abi.encodePacked(address(this), block.timestamp));
+
+        // Create Safe Guard instance and write it into its designated slot
+        // solhint-disable-next-line no-inline-assembly
+        assembly {
+            guard := create2(0, add(bytecode, 0x20), mload(bytecode), salt)
+        }
```

Comment:
Safe Guard instance is created, and the last multisig owner is set as the created Safe Guard owner.